crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -964,8 +964,8 @@ impl Backend {
             // Defaults to block number for compatibility with existing state files.
             let fork_num_and_hash = self.get_fork().map(|f| (f.block_number(), f.block_hash()));
 
+            let best_number = state.best_block_number.unwrap_or(block.number.to::<U64>());
```

Comment:
this looks right, this only moved this out of the if else where we had this in both branches